### PR TITLE
optional parameter force_turn_costs

### LINKED ISSUE #1220: Configuration option to force routing to select Turn Cost sensitive algo for routing
A customer wants to have the option to have a profile have CH enabled (for matrix) but still have routing respect turn costs. Algorithm selection needs to default to `FLEX_PREPROCESSED` mode instead of `FLEX_STATIC`. 

Easiest place to put is `encoder_options` of the respective profile, where we also enable turn costs, e.g. 
```
  "profile-car": {
    "profiles": "driving-car",
    "parameters": {
      "encoder_options": "turn_costs=true|force_turn_costs=true|...",
      ...
```

We decided to have a patch release based off of the last release, although this breaks our version semantics a bit.

### PR DESCRIPTION
**Pull request by takb**
### Pull Request Checklist

- [ ] 1. I have [**rebased**][rebase] the latest version of the master branch into my feature branch and all conflicts
         have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes #1220 .

### Information about the changes
- Key functionality added:
optional profile parameter `force_turn_costs`
If `true`, algorithm selection defaults to `FLEX_PREPROCESSED` mode instead of `FLEX_STATIC`.

### CODE REVIEW

**Review by sfendrich — APPROVED** [review]
LGTM

**Review by aoles — CHANGES_REQUESTED** [review]
Thanks a lot @takb ! 

I encourage you to re-consider whether the new parameter `force_turn_cost` really belongs under `encoder_options`. I would argue that the parameters to  `encoder_options` influence the base graph storage at build time, whereas forcing turn costs is a strictly run-time parameter. Therefore, my suggestion would be to move the parameter one level up to the general profile parameters. Does that make sense to you? Cheers!

**Comment by takb** [comment]
You're absolutely right @aoles , moved the parameter.

File: CHANGELOG.md [hunk 1]
```diff
@@ -59,6 +59,10 @@ RELEASING:
 - fix minor spelling errors in Places.md ([#1196](https://github.com/GIScience/openrouteservice/issues/1196))
 - address matrix failures for HGV profile ([#1198](https://github.com/GIScience/openrouteservice/issues/1198))
 
+## [6.7.1] - 2022-10-05
+### Added
+- optional `encoder_options` parameter `force_turn_costs` ([#1220](https://github.com/GIScience/openrouteservice/pull/1220))
```
Comment (aoles) [inline CHANGELOG.md]: Needs updating after the recent changes.

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/isochrones/fast/ResultTest.java [hunk 2]
```diff
@@ -380,7 +380,7 @@ public void testSmoothingFactor() {
                 .then()
                 .body("any { it.key == 'type' }", is(true))
                 .body("any { it.key == 'features' }", is(true))
-                .body("features[0].geometry.coordinates[0].size", is(69))
+                .body("features[0].geometry.coordinates[0].size()", is(69))
```
Comment (aoles) [inline openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/isochrones/fast/ResultTest.java:383]: Could you maybe briefly clarify, how this change relates to the PR?

**Review by aoles — CHANGES_REQUESTED** [review]
Thanks @takb , please see my inline comments. Cheers!

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/isochrones/fast/ResultTest.java (hunk 2 above)
Comment (takb) [reply openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/isochrones/fast/ResultTest.java:383]: not at all, but the notation without the parentheses causes problems when run on newer version JVMs

**Review by aoles — APPROVED** [review]
Thanks again @takb !

**Pull request merged**